nav2_waypoint_follower/src/waypoint_follower.cpp

```diff
@@ -304,4 +329,174 @@ WaypointFollower::goalResponseCallback(
+  {
+    RCLCPP_ERROR(get_logger(), "Send goal call failed");
+    return;
+  }
+  // Get the goal handle and save so that we can check
+  // on completion in the timer callback
+  waypoint_follower_goal_handle_ = future_goal_handle.get();
+  if (!waypoint_follower_goal_handle_) {
+    RCLCPP_ERROR(get_logger(), "Goal was rejected by server");
+    return;
+  }
+}
+
+std::vector<geometry_msgs::msg::PoseStamped>
+WaypointFollower::convertGPSWaypointstoPosesinMap(
+  const
+  std::vector<sensor_msgs::msg::NavSatFix> & gps_waypoints,
+  const rclcpp_lifecycle::LifecycleNode::SharedPtr & parent_node,
+  const rclcpp::Client<robot_localization::srv::FromLL>::SharedPtr & fromll_client)
+{
+  RCLCPP_INFO(parent_node->get_logger(), "Converting GPS waypoints to Map Frame..");
+  if (!parent_node) {
+    throw std::runtime_error{
+            "Failed to lock node while trying to convert GPS waypoints to Map frame"};
+  }
+  std::vector<geometry_msgs::msg::PoseStamped> poses_in_map_frame_vector;
+  int index_of_gps_waypoints = 0;
+  for (auto && curr_gps_waypoint : gps_waypoints) {
+    auto fromLLRequest = std::make_shared<robot_localization::srv::FromLL::Request>();
+    fromLLRequest->ll_point.latitude = curr_gps_waypoint.latitude;
+    fromLLRequest->ll_point.longitude = curr_gps_waypoint.longitude;
+    fromLLRequest->ll_point.altitude = curr_gps_waypoint.altitude;
+    if (!fromll_client->wait_for_service((std::chrono::seconds(10)))) {
+      RCLCPP_ERROR(
+        parent_node->get_logger(),
+        "fromLL service from robot_localization is not available"
+        "cannot convert GPS waypoints to Map frame poses,"
+        "Make sure you have run navsat_transform_node of robot_localization");
+      return std::vector<geometry_msgs::msg::PoseStamped>();
+    }
+    RCLCPP_INFO(
+      parent_node->get_logger(), "Processing for %i'th Waypoint..",
+      index_of_gps_waypoints);
+    auto inner_client_callback =
+      [&, parent_node](rclcpp::Client<robot_localization::srv::FromLL>::SharedFuture inner_future)
+      {
+        auto result = inner_future.get();
+      };
+    auto inner_future_result = fromll_client->async_send_request(
+      fromLLRequest,
+      inner_client_callback);
+    // this poses are assumed to be on global frame (map)
+    geometry_msgs::msg::PoseStamped curr_waypoint_in_map_frame;
+    curr_waypoint_in_map_frame.header.frame_id = "map";
+    curr_waypoint_in_map_frame.header.stamp = rclcpp::Clock().now();
+    curr_waypoint_in_map_frame.pose.position.x = inner_future_result.get()->map_point.x;
+    curr_waypoint_in_map_frame.pose.position.y = inner_future_result.get()->map_point.y;
+    curr_waypoint_in_map_frame.pose.position.z = inner_future_result.get()->map_point.z;
+
+    tf2::Quaternion quat_tf;
+    quat_tf.setRPY(0, 0, 0);
+    geometry_msgs::msg::Quaternion quat_msg;
+    tf2::convert(quat_msg, quat_tf);
+    curr_waypoint_in_map_frame.pose.orientation = quat_msg;
+
+    RCLCPP_INFO(
```

Comment:
TMI, lets remove this log. It was probably really helpful for development, but a little much for average users.